examples/experimental/dagster-airlift/dagster_airlift/in_airflow/base_asset_operator.py

```diff
@@ -141,40 +139,39 @@ def launch_runs_for_task(self, context: Context, dag_id: str, task_id: str) -> N
             )
-        logger.info(f"Found {len(assets_to_trigger_per_job)} jobs to trigger")
-
-        triggered_runs = []
-        for job_identifier, asset_key_paths in assets_to_trigger_per_job.items():
-            logger.info(f"Triggering run for {job_identifier} with assets {asset_key_paths}")
-            run_id = self.launch_dagster_run(
-                context,
-                session,
-                dagster_url,
-                _build_dagster_run_execution_params(
-                    self.default_dagster_run_tags(context),
-                    job_identifier,
-                    asset_key_paths,
-                ),
+            == 1
+        ):
+            raise Exception(
+                "Could not find an implicit asset job that can trigger all assets in this task. "
+                "This may mean that you need to upgrade your Dagster version (1.8 or later), which allows all assets to be materialized in a single run, "
+                "or that the assets are not in the same code location. "
+                "`dagster-airlift` expects that all assets mapped to a given task exist within the same code location, so that they can be executed by the same run."
             )
-            triggered_runs.append(run_id)
-        completed_runs = {}  # key is run_id, value is status
-        while len(completed_runs) < len(triggered_runs):
-            for run_id in triggered_runs:
-                if run_id in completed_runs:
-                    continue
-                run_status = self.get_dagster_run_status(session, dagster_url, run_id)
-                if run_status in ["SUCCESS", "FAILURE", "CANCELED"]:
-                    logger.info(f"Run {run_id} completed with status {run_status}")
-                    completed_runs[run_id] = run_status
-        non_successful_runs = [
-            run_id for run_id, status in completed_runs.items() if status != "SUCCESS"
-        ]
-        if non_successful_runs:
-            raise Exception(f"Runs {non_successful_runs} did not complete successfully.")
+
+        job_identifier = _get_implicit_job_identifier(next(iter(filtered_asset_nodes)))
+        asset_key_paths = [asset_node["key"]["path"] for asset_node in filtered_asset_nodes]
+        logger.info(f"Triggering run for {job_identifier} with assets {asset_key_paths}")
+        run_id = self.launch_dagster_run(
+            context,
+            session,
+            dagster_url,
+            _build_dagster_run_execution_params(
+                self.default_dagster_run_tags(context),
+                job_identifier,
+                asset_key_paths=[asset_node["key"]["path"] for asset_node in filtered_asset_nodes],
+            ),
+        )
+        logger.info("Waiting for dagster run completion...")
+        while status := self.get_dagster_run_status(session, dagster_url, run_id):
+            if status in ["SUCCESS", "FAILURE", "CANCELED"]:
+                break
+            time.sleep(5)
```

Comment:
how did you choose five seconds?

Reply:
This looks new?

Reply:
Yea it's new. Somewhat obfuscated by the previous logic but we're slamming the graphql API in search of status. I picked 5 seconds somewhat arbitrarily, but seemed like it wouldn't rate limit us

Reply:
5 seconds seems a little much. How about 1 second and make it configurable?